src/api.jl

```diff
@@ -0,0 +1,24 @@
+module api
+
+paths = ByteString[]
+const libvulkan = Libdl.find_library(["libvulkan", "vulkan", "libvulkan.so.1"], paths)
+@assert libvulkan != ""
+
+function _vk_func(func, ret_type, arg_types)
+  local args_in = Symbol[symbol("arg$i::$T")
+  for (i, T) in enumerate(arg_types.args)]
+  esc(quote
+    function $func($(args_in...))
+      ccall(($(string(func)), libvulkan),
+      $ret_type,
+      $arg_types,
+      $(args_in...))
+    end
+  end)
+end
```

Comment:
From providing a library perspective I would like to have the method definition to be as tight as possible, to prevent errors.

So I take it I need to add `cconvert` and `unsafe_convert` for all kinds of types we might pass as references? I thought as long as all types involved are `isbits` Julia should be able to do the transformation automatically.